PROTOCOL.md

```diff
@@ -574,7 +574,7 @@ The following is an example `rowIdHighWaterMark` action:
 ```
 
 ### Domain Metadata
-The domain metadata action contains a configuration (string-string map) for a named metadata domain. Two overlapping transactions conflict if they both contain a domain metadata action for the same metadata domain.
+The domain metadata action contains a configuration (JSON string) for a named metadata domain. Two overlapping transactions conflict if they both contain a domain metadata action for the same metadata domain.
```

Comment:
Should the spec actually _require_ a JSON string? Or just say it's a string and let each domain choose whether it's a plain string, JSON string, base64/base85 encoded binary blob, etc? 

That _would_ make it harder to guess how a given string should be handled, but:
1. Presumably anyone who tries to parse a given domain configuration should already know what to expect.
2. JSON strings can still lead to parsing problems if e.g. reader expected an int value and finds an object instead. If we _really_ want to ensure domain configs are universally parseable, we would have to go back to string-string map.

Reply:
I'd say just use a string. Using a JSON object with fields is then simply a best practice for extensibility.

Reply:
agreed